lib/transmuxer/h265.js

```diff
@@ -0,0 +1,855 @@
+    const subLayerLevelPresentFlag = [];
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      subLayerProfilePresentFlag.push(gb.readBoolean());
+      subLayerLevelPresentFlag.push(gb.readBoolean());
+    }
+    if (maxSubLayersMinus1 > 0) {
+      for (let i = maxSubLayersMinus1; i < 8; i++) {
+        gb.readBits(2);
+      }
+    }
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      if (subLayerProfilePresentFlag[i]) {
+        gb.readBits(88);
+      }
+      if (subLayerLevelPresentFlag[i]) {
+        gb.readUnsignedByte();
+      }
+    }
+    // profile_tier_level end
+
+    gb.readUnsignedExpGolomb(); // seq_parameter_set_id
+    const chromaFormatIdc = gb.readUnsignedExpGolomb();
+    if (chromaFormatIdc == 3) {
+      gb.readBits(1); // separate_colour_plane_flag
+    }
+    const picWidthInLumaSamples = gb.readUnsignedExpGolomb();
+    const picHeightInLumaSamples = gb.readUnsignedExpGolomb();
+    const conformanceWindowFlag = gb.readBoolean();
+    if (conformanceWindowFlag) {
+      leftOffset += gb.readUnsignedExpGolomb();
+      rightOffset += gb.readUnsignedExpGolomb();
+      topOffset += gb.readUnsignedExpGolomb();
+      bottomOffset += gb.readUnsignedExpGolomb();
+    }
+    const bitDepthLumaMinus8 = gb.readUnsignedExpGolomb();
+    const bitDepthChromaMinus8 = gb.readUnsignedExpGolomb();
+    const log2MaxPicOrderCntLsbMinus4 = gb.readUnsignedExpGolomb();
+    const subLayerOrderingInfoPresentFlag = gb.readBoolean();
+    for (
+      let i = subLayerOrderingInfoPresentFlag ? 0 : maxSubLayersMinus1;
+      i <= maxSubLayersMinus1;
+      i++
+    ) {
+      gb.readUnsignedExpGolomb(); // max_dec_pic_buffering_minus1[i]
+      gb.readUnsignedExpGolomb(); // max_num_reorder_pics[i]
+      gb.readUnsignedExpGolomb(); // max_latency_increase_plus1[i]
+    }
+    gb.readUnsignedExpGolomb(); // log2_min_luma_coding_block_size_minus3
+    gb.readUnsignedExpGolomb(); // log2_diff_max_min_luma_coding_block_size
+    gb.readUnsignedExpGolomb(); // log2_min_transform_block_size_minus2
+    gb.readUnsignedExpGolomb(); // log2_diff_max_min_transform_block_size
+    gb.readUnsignedExpGolomb(); // max_transform_hierarchy_depth_inter
+    gb.readUnsignedExpGolomb(); // max_transform_hierarchy_depth_intra
+    const scalingListEnabledFlag = gb.readBoolean();
+    if (scalingListEnabledFlag) {
+      const spsScalingListDataPresentFlag = gb.readBoolean();
+      if (spsScalingListDataPresentFlag) {
+        for (let sizeId = 0; sizeId < 4; sizeId++) {
+          for (
+            let matrixId = 0;
+            matrixId < (sizeId === 3 ? 2 : 6);
+            matrixId++
+          ) {
+            const scalingListPredModeFlag = gb.readBoolean();
+            if (!scalingListPredModeFlag) {
+              gb.readUnsignedExpGolomb(); // scaling_list_pred_matrix_id_delta
+            } else {
+              const coefNum = Math.min(64, 1 << (4 + (sizeId << 1)));
+              if (sizeId > 1) {
```

Comment:
Same here

Reply:
The code of sps parser is based on mpegts.js (see: https://github.com/xqq/mpegts.js/blob/master/src/demux/h265-parser.js#L65), so I'll add a comment , because I don't know why theses values